File: software/examples/ruby/example_callback.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_motion_detector'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change XYZ to the UID of your Motion Detector Bricklet

ipcon = IPConnection.new # Create IP connection
md = BrickletMotionDetector.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Register motion detected callback
md.register_callback(BrickletMotionDetector::CALLBACK_MOTION_DETECTED) do
  puts 'Motion Detected'
end

# Register detection cycle ended callback
md.register_callback(BrickletMotionDetector::CALLBACK_DETECTION_CYCLE_ENDED) do
  puts 'Detection Cycle Ended (next detection possible in ~3 seconds)'
end

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect

File: software/examples/csharp/ExampleCallback.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your Motion Detector Bricklet

	// Callback function for motion detected callback
	static void MotionDetectedCB(BrickletMotionDetector sender)
	{
		Console.WriteLine("Motion Detected");
	}

	// Callback function for detection cycle ended callback
	static void DetectionCycleEndedCB(BrickletMotionDetector sender)
	{
		Console.WriteLine("Detection Cycle Ended (next detection possible in ~3 seconds)");
	}

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletMotionDetector md = new BrickletMotionDetector(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Register motion detected callback to function MotionDetectedCB
		md.MotionDetectedCallback += MotionDetectedCB;

		// Register detection cycle ended callback to function DetectionCycleEndedCB
		md.DetectionCycleEndedCallback += DetectionCycleEndedCB;

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}
